application_store/_helpers/application.py

```diff
@@ -19,3 +33,57 @@ def order_applications(applications, order_by, order_rev):
             reverse=int(order_rev),
         )
     return applications
+
+
+def send_submit_notification(
+    application_with_form_json,
+    eoi_results,
+    account,
+    application_with_form_json_and_fund_name,
+    application,
+    round_data,
+):
+    contents = {
+        NotifyConstants.APPLICATION_FIELD: application_with_form_json_and_fund_name,
+        NotifyConstants.MAGIC_LINK_CONTACT_HELP_EMAIL_FIELD: round_data.contact_email,
+    }
+
+    if eoi_results:
+        eoi_decision = eoi_results["decision"]
+        if eoi_decision == Decision.FAIL:
+            return  # we don't send emails for a failure
+
+        full_name = (
+            account.full_name
+            if account.full_name
+            else map_application_key_fields(
+                application_with_form_json,
+                ROUND_ID_TO_KEY_REPORT_MAPPING[application.round_id],
+                application.round_id,
+            ).get("lead_contact_name", "")
+        )
+
+        if Decision(eoi_decision) == Decision.PASS:  # EOI Full pass
+            notify_template = Config.NOTIFY_TEMPLATE_EOI_PASS
+
+        elif Decision(eoi_decision) == Decision.PASS_WITH_CAVEATS:  # EOI Pass with caveats
+            notify_template = Config.NOTIFY_TEMPLATE_EOI_PASS_W_CAVEATS
+            contents = [NotifyConstants.APPLICATION_CAVEATS] = (eoi_results["caveats"],)
+    else:
+        notify_template = Config.NOTIFY_TEMPLATE_SUBMIT_APPLICATION
+        eoi_decision = None
+        full_name = account.full_name
+
+    contents["application"] = create_qa_base64file(contents.get("application"), True)
+    del contents["application"]["forms"]
```

Comment:
Rather than mutating here could we return the thing we want directly e.g.
`application_contents = create_qa_base64file(contents.get("application"), True)` and pass that to the `Notification`?

Reply:
Have tweaked this slightly but reluctant to change any further as it's feeling less like a shift and shift and that function is used in other places.